csrc/cpu/neighbor_sample_cpu.cpp

```diff
@@ -223,7 +245,8 @@ hetero_sample(const vector<node_t> &node_types,
       const auto &begin = slice_dict.at(dst_node_type).first;
       const auto &end = slice_dict.at(dst_node_type).second;
       for (int64_t i = begin; i < end; i++) {
-        const auto &w = dst_samples[i];
+        const auto &w = temporal ? temp_dst_samples[i].first : dst_samples[i];
+        const int64_t source_w = temporal ? temp_dst_samples[i].second : -1;
```

Comment:
why is it called source_w? Is it the root node?

Reply:
yes, do you prefer rename it to `root_w`?

Reply:
sounds good